src/js/view/view.js

```diff
@@ -807,12 +842,41 @@ function View(_api, _model) {
     };
 
     this.setIntersection = function (entry) {
-        _model.set('intersectionRatio', entry.intersectionRatio);
+        if (entry.target === _playerElement) {
+            _model.set('intersectionRatio', entry.intersectionRatio);
+        } else if (entry.target === _containerElement) {
+            _setFloatingIntersection(entry);
+        }
     };
 
+    function _setFloatingIntersection(entry) {
+        // Entirely invisible and no floating player already in the DOM
+        const isVisible = entry.intersectionRatio === 1;
+        if (!isVisible && _model.get('state') !== STATE_IDLE && floatingPlayer === null) {
+            floatingPlayer = _containerElement;
+
+            const width = _containerElement.offsetWidth;
+            const height = _containerElement.offsetHeight;
+
+            addClass(_playerElement, 'jw-flag-floating');
+            _this.trigger(FLOAT, { floating: true });
+
+            _this.resize(320, 320 * height / width);
+            style(_containerElement, { width, height }); // Keep original dimensions.
```

Comment:
This creates problems when using percentages. `Width` and `height` of the wrapper and the player need to match. Setting `width: 50%` causes the wrapper to be 50% of its container, and also causes the player to be 50% of the wrapper. Rather than having a div wrap around the player, try using one that wraps around the contents _inside_ the player div: https://github.com/jwplayer/jwplayer/pull/3043